test_conformance/subgroups/test_workgroup_non_uniform_vote.cpp

```diff
@@ -0,0 +1,504 @@
+                                       non_uniform_all_equal_source, 0,
+                                       useCoreSubgroups_, required_extensions_);
+        return error;
+    }
+
+    template <typename T, unsigned int work_items_mask> int run_elect()
+    {
+        int error =
+            test<T, ELECT<work_items_mask>, GWS_NON_UNIFORM,
+                 LWS_NON_UNIFORM>::run(device_, context_, queue_, num_elements_,
+                                       "test_elect", elect_source, 0,
+                                       useCoreSubgroups_, required_extensions_,
+                                       static_cast<int>(work_items_mask));
+
+        return error;
+    }
+
+    template <typename T> int run_nu_any()
+    {
+        int error =
+            test<T, AAN<T, NonUniformVoteOp::any>, GWS_NON_UNIFORM,
+                 LWS_NON_UNIFORM>::run(device_, context_, queue_, num_elements_,
+                                       "test_non_uniform_any",
+                                       non_uniform_any_source, 0,
+                                       useCoreSubgroups_, required_extensions_);
+        return error;
+    }
+
+    template <typename T> int run_nu_all()
+    {
+        int error =
+            test<T, AAN<T, NonUniformVoteOp::all>, GWS_NON_UNIFORM,
+                 LWS_NON_UNIFORM>::run(device_, context_, queue_, num_elements_,
+                                       "test_non_uniform_all",
+                                       non_uniform_all_source, 0,
+                                       useCoreSubgroups_, required_extensions_);
+        return error;
+    }
+
+
+private:
+    cl_device_id device_;
+    cl_context context_;
+    cl_command_queue queue_;
+    int num_elements_;
+    bool useCoreSubgroups_;
+    std::vector<std::string> required_extensions_;
+};
+}
+
+int test_work_group_functions_non_uniform_vote(cl_device_id device,
+                                               cl_context context,
+                                               cl_command_queue queue,
+                                               int num_elements)
+{
+    std::vector<std::string> required_extensions = {
+        "cl_khr_subgroup_non_uniform_vote"
+    };
+    run_for_type rft(device, context, queue, num_elements, true,
+                     required_extensions);
+
+    int error = rft.run_nu_all_eq<cl_char>();
+    error |= rft.run_nu_all_eq<cl_uchar>();
+    error |= rft.run_nu_all_eq<cl_short>();
+    error |= rft.run_nu_all_eq<cl_ushort>();
+    error |= rft.run_nu_all_eq<cl_int>();
+    error |= rft.run_nu_all_eq<cl_uint>();
+    error |= rft.run_nu_all_eq<cl_long>();
+    error |= rft.run_nu_all_eq<cl_ulong>();
+    error |= rft.run_nu_all_eq<cl_float>();
+    error |= rft.run_nu_all_eq<cl_double>();
+    error |= rft.run_nu_all_eq<subgroups::cl_half>();
+    error |= rft.run_elect<int, 0xffffaaaa>();
+    error |= rft.run_elect<int, 0x00000000>();
```

Comment:
The mask of 0 here is pointless, as the kernel will not call `sub_group_elect()` at all, resulting in unnecessary noise in the log output.

Reply:
It _would_ make sense to test with all work items active, but as has been said, there can be more than 32 work items in a subgroup.

Reply:
I think kernel is prepared in such way that each workgroup size will be properly handled.

Reply:
Oh, I will have to make a correction according to comment https://github.com/KhronosGroup/OpenCL-CTS/pull/1074#discussion_r576880173

Reply:
If you're assuming GitHub's UI gives any indication as to the exact comment you've replied to, please note that it does not - please quote comments if (and only if) there is ambiguity (or some other lack of clarity) as to what you're replying to.

> I think kernel is prepared in such way that each subgroup size will be properly handled.

I assume this in response to my question about the mask of 0x00000000.  Are you working under the false impression that this works in the way that a mask of 0xffffffff would actually work?  (Or am I perhaps mistaken about this?)

Reply:
> that this works in the way that a mask of 0xffffffff would actually work? (Or am I perhaps mistaken about this?)

I mean that fixing existing kernel it will work for any subgroup size, then testing mask 0xffffffff will work correctly